### DISCUSSION TITLE
MONAILabel / 3D Slicer with Google Colab?

### DISCUSSION THREAD

**Discussion by ChrisPlatais** [Project-MONAI/MONAILabel#709] [Q&A] [answered]
Hello,

I'm a clinician and currently in the process of setting up a project that involves segmentation of 2-dimensional panoramic radiographs. The plan would be for a number of radiologists to work on segmenting a database of anonymised images using the MONAILabel extension of 3D Slicer. To my knowledge, our most convenient departmental PCs do not have CUDA-capable GPUs. My colleagues in Imaging Sciences have suggested that Google Colab could be a solution - is it possible to run a MONAILabel server with Google Colab?

Thank you.

Chris

**Comment by diazandr3s** [comment]
Hi @ChrisPlatais,


Thanks for opening this discussion. 
Currently, MONAI Label Apps are focused on 3D medical images and histopathology applications.
We haven't developed a 2D Segmentation App yet. But it'll be interesting to know more about this use case. 
Regarding Colab, I think it is a great tool to train and test deep learning models. But it is limited in time and available resources.

Please let me know and we can arrange a meeting to talk more about this use case.


Andres

**Comment by lassoan** [comment]
> most convenient departmental PCs do not have CUDA-capable GPUs

Each client computer needs to be able to run Slicer but does not need a GPU, as all clients can connect to a single shared MONAILabel server.

The most cost-effective "server" is probably a desktop gaming computer with a GeForce RTX 3090, which you can get for about $5k. Alternatively, you could get a desktop workstation computer with an NVIDIA RTX6000 for a few k$ more. Many hospitals set up deals with Google/Amazon/Microsoft for getting GPU computation resources for "free" (in return for access to patient data), and more and more hospitals set up shared GPU servers locally, so before you buy anything you may ask IT folks what is available for you already.

**Comment by SachidanandAlle** [comment, marked as answer]
You can run MONAILabel in Google Colab.  But you would like to make it persistent, continuous and scalable solution for the use-case.. MONAILabel server is not just limited to train a model but more for the users to run annotations using x/y/z clients/viewers.  There are lot more associations (using different datastore like filesystem, http based storage stystem, dicomweb, xnat etc..) to which MONAILabel server has to connect.. all these integrations to get exposed through Colab is little tricky and we might end up with some kind of hacks.

So, I would recommend not to think in that direction (running Colab/notebook etc..) and consider it as more of e2e deployable solution.
